packages/visual-group/src/group-helper/matrix-model.js

```diff
@@ -248,6 +248,20 @@ export const getMatrixModel = (dataModel, facetsAndProjections, valueCellCreator
         });
     }
 
+    const colFacetFormatters = colFacets.map(facetField => facetField.rawFormat());
+    columnKeys.forEach((rColumnKeys) => {
+        rColumnKeys.forEach((columnKey, idx) => {
+            rColumnKeys[idx] = colFacetFormatters[idx](columnKey);
+        });
+    });
+
+    const rowFacetFormatters = rowFacets.map(facetField => facetField.rawFormat());
+    rowKeys.forEach((rRowKeys) => {
+        rRowKeys.forEach((rowKey, idx) => {
```

Comment:
Functions for column and row are the same, use pure function and call for both row and column